src/cgame/cg_rocket_draw.cpp

```diff
@@ -1029,6 +1031,36 @@ class WallwalkElement : public HudElement
 	bool isActive;
 };
 
+class SprintElement : public HudElement
+{
+public:
+	SprintElement( const Rocket::Core::String& tag ) :
+			HudElement( tag, ELEMENT_HUMANS )
+	{
+		SetActive( false );
+	}
+
+	void DoOnUpdate() override
+	{
+		bool sprinting = cg.snap->ps.stats[ STAT_STATE ] & SS_SPEEDBOOST;
+		if ( sprinting != isActive )
+		{
+			SetActive( sprinting );
+		}
+	}
+
+private:
+	void SetActive( bool active )
+	{
+		isActive = active;
+		SetClass( "active", active );
+		SetClass( "inactive", !active );
```

Comment:
Can you change the name to something more understandable? Maybe something like `running` and `sprinting`?

Reply:
Ideally, I would prefer to merge this class and the wallwalking one, to avoid code duplication. To some extent, it could also be used for no-clip weapons (to disable showing the clip) and other elements like that, and renaming the "vars" would not help at that.

Reply:
You could have a virtual function that returns a string for what CSS class to use. Store that instead of a bool, and then you could use any names and it generalizes to more than two states.

Reply:
yes, that would likely be the best solution, but I'm not really fond of the idea of playing too much with librocket. The current situation seems buggy to me (see #1651) but I am clueless about how to really fix it.